File: client/src/core/types/schema.ts
export type ApiCoreColumnKind = 'numeric' | 'categorical' | 'date';

export interface ApiCoreDatasetColumnInfo {
    name: string;
    kind: ApiCoreColumnKind;
    unique_values?: string[] | null;
    unique_count?: number | null;
    numeric_min?: number | null;
    numeric_max?: number | null;
    date_min?: string | null;
    date_max?: string | null;
}

export interface ApiCoreDatasetSchemaResults {
    dataset_name: string;
    columns: ApiCoreDatasetColumnInfo[];
    max_unique_values: number;
}

File: app/core/models/schema.py
from __future__ import annotations

from enum import StrEnum

from pydantic import BaseModel, ConfigDict


class ApiCoreColumnKind(StrEnum):
    NUMERIC = "numeric"
    CATEGORICAL = "categorical"
    DATE = "date"


class ApiCoreDatasetColumnInfo(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: str
    kind: ApiCoreColumnKind

    unique_values: list[str] | None = None
    unique_count: int | None = None

    numeric_min: float | None = None
    numeric_max: float | None = None

    date_min: str | None = None
    date_max: str | None = None


class ApiCoreDatasetSchemaResults(BaseModel):
    model_config = ConfigDict(extra="forbid")

    dataset_name: str
    columns: list[ApiCoreDatasetColumnInfo]
    max_unique_values: int

